src/paho/mqtt/client.py

```diff
@@ -2671,6 +2694,89 @@ def _reconnect_wait(self):
             time.sleep(min(remaining, 1))
             remaining = target_time - time_func()
 
+    def _get_proxies(self):
+        def proxy_is_valid(p):
+            if isinstance(p, dict):
+                return bool(p.get("proxy_type"))
+            elif isinstance(p, (list, tuple)):
+                return len(p) == 6 and bool(p[0])
+            else:
+                return False
+
+        proxies = []  # proxy servers to use in order of preference
+
+        if proxy_is_valid(self._proxy):
+            proxies.append(self._proxy)
+
+        socks_default = socks.get_default_proxy()
+        if proxy_is_valid(socks_default):
+            proxies.append({
+                "proxy_type": socks_default[0],
+                "proxy_addr": socks_default[1],
+                "proxy_port": socks_default[2],
+                "proxy_rdns": socks_default[3],
+                "proxy_username": socks_default[4],
+                "proxy_password": socks_default[5]
+            })
+
+        # Finally, look at environment variables ending in _proxy as long as the
+        # host we're trying to connect to isn't listed under the no_proxy
+        # environment variable (matches built-in module urllib's behavior)
+        if not urllib_dot_request.proxy_bypass(self._host):
+            env_proxies = urllib_dot_request.getproxies()
+            for proxy in env_proxies.values():
```

Comment:
If I understand well, this will catch an "http_proxy" or "https_proxy" to proxy MQTT protocol. Shouldn't we use an "mqtt_proxy" variable ?

The other question I have is how to disable proxying. Let's imaging my program doing both http request and use paho-mqtt. How to tell it to use the proxy for http but not mqtt ?

Reply:
Great catches for both of these. I will change the code to use an `mqtt_proxy` variable only. For the second question, what if we add a `no_proxy` flag to `connect()` and related methods that defaults to True but, if set to False, bypasses all proxies for MQTT connections?